runtime/amplitude/src/xcm_config.rs

```diff
@@ -166,11 +91,11 @@ pub type LocalAssetTransactor = MultiCurrencyAdapter<
 	// Use this fungibles implementation
 	Currencies,
 	(), // We don't handle unknown assets.
-	IsNativeConcrete<CurrencyId, CurrencyIdConvert>,
+	IsNativeConcrete<CurrencyId, CurrencyIdConvert<ParachainInfo, AssetRegistryInspector>>,
```

Comment:
So the naming of the generic `CurrencyIdConvert` is a bit unfortunate, because it should be `AssetRegisterInspector` to be more clear. 

At first I attempted to restrict the second generic argument to an instance of the asset registry itself, but turns out that the methods that we need from the asset registry are not implemented on a specific trait, rather the pallet itself. For instance, [here](https://github.com/pendulum-chain/pendulum/pull/432/files#diff-3287a4ee060b1070a8e6559a094b4890a596a1b281effef54cafe59da20edf49R239) we call the getter for the metadata.

Because of this restriction, I ended up defining this `AssetRegistryInspector` which is just a [wrapper](https://github.com/pendulum-chain/pendulum/pull/432/files#diff-4e8d76261f23472a35cee760b0818583d9267e55b361e44f2c646ab1c031d7d0R64) around the methods we require. I couldn't find a way to make the compiler happy and accept a generic argument that is of type `AssetRegistry`, and it makes sense (I think) since the methods are not on a trait.

Reply:
I see, makes sense to create the extra trait if something like it doesn't exist yet. Thanks for explaining.

Reply:
@gianfra-t I think it is possible, see my other comment.

Reply:
I have simplified the code, thanks for the suggestion!